### PR TITLE
fix: adjusts form redirects with updated useRouter

### PR DESCRIPTION
**Pull request by PatrikKozak**

### CODE REVIEW

File: src/components/CMSForm/index.tsx [hunk 1]
```diff
@@ -82,7 +82,14 @@ export const CMSForm: React.FC<{
           setHasSubmitted(true)
 
           if (confirmationType === 'redirect' && redirect) {
-            if (redirect.url) router.push(redirect.url)
+            const { url } = redirect
+
+            const redirectUrl = new URL(url)
```
Comment (jacobsfletch) [inline src/components/CMSForm/index.tsx:87]: Does this safely fail? Probably want to wrap with a condition if `url` exists then also a `try...catch`

Comment (PatrikKozak) [reply src/components/CMSForm/index.tsx]: @jacobsfletch This has been added.

File: src/components/CMSForm/index.tsx [hunk 2]
```diff
@@ -82,7 +82,23 @@ export const CMSForm: React.FC<{
...
 
           if (confirmationType === 'redirect' && redirect) {
-            if (redirect.url) router.push(redirect.url)
+            const { url } = redirect
+
+            try {
+              if (url) {
+                const redirectUrl = new URL(url)
+
+                if (url && redirectUrl.origin === process.env.NEXT_PUBLIC_APP_URL) {
```
[the remark this answers is not in the document]
Comment (PatrikKozak) [reply src/components/CMSForm/index.tsx]: @jacobsfletch Done.

File: src/components/CMSForm/index.tsx [hunk 3]
```diff
@@ -82,7 +82,23 @@ export const CMSForm: React.FC<{
...
+              const redirectUrl = new URL(url)
+
+              if (redirectUrl.origin === process.env.NEXT_PUBLIC_APP_URL) {
+                router.push(redirectUrl.href)
+              }
+              window.location.assign(url)
+            } catch (err) {
+              console.warn(err)
+              setError({
+                message: 'Somethign went wrong. Did not redirect.',
```
Comment (thgh) [inline src/components/CMSForm/index.tsx:99]:
```suggestion
                message: 'Something went wrong. Did not redirect.',
```
Also, translations?

File: src/components/CMSForm/index.tsx [hunk 4]
```diff
@@ -82,7 +82,23 @@ export const CMSForm: React.FC<{
           setHasSubmitted(true)
 
           if (confirmationType === 'redirect' && redirect) {
-            if (redirect.url) router.push(redirect.url)
+            const { url } = redirect
+
+            if (!url) return
```
Comment (thgh) [inline src/components/CMSForm/index.tsx:87]: returning here will skip the `submitForm()` at the end, is this intended? (maybe add a comment for that)

File: src/components/CMSForm/index.tsx (hunk 3 above)
Comment (jacobsfletch) [reply src/components/CMSForm/index.tsx:99]: @thgh nice catch

**Pull request merged**